app/src/main/java/io/apicurio/registry/rules/validation/OpenApiContentValidator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.rules.validation;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.apicurio.datamodels.Library;
+import io.apicurio.datamodels.core.models.Document;
+import io.apicurio.datamodels.core.models.ValidationProblem;
+
+/**
+ * A content validator implementation for the OpenAPI content type.
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+public class OpenApiContentValidator implements ContentValidator {
+
+    /**
+     * @see io.apicurio.registry.rules.validation.ContentValidator#validate(io.apicurio.registry.rules.validation.ValidationLevel, java.lang.String)
+     */
+    @Override
+    public void validate(ValidationLevel level, String artifactContent) throws InvalidContentException {
```

Comment:
If the level is null, the validator would check syntax only and won't raise a exception

Reply:
OK fair enough.